### PR TITLE
Move setupRootAcl from AuthManager to TestSubject

### PR DESCRIPTION
**Pull request by edwardsph**
This is a small refactor to move a function to a more appropriate location. It also involved converting the static ClientRegistry class to an ApplicationScoped bean which simplified testing.

### CODE REVIEW

File: src/main/java/org/solid/testharness/http/SolidClient.java [hunk 1]
```diff
@@ -30,10 +32,15 @@
...
+        final ClientRegistry clientRegistry = CDI.current().select(ClientRegistry.class).get();
+        client = clientRegistry.getClient(ClientRegistry.DEFAULT);
     }
     public SolidClient(final String user) {
-        client = ClientRegistry.getClient(user);
+        final ClientRegistry clientRegistry = CDI.current().select(ClientRegistry.class).get();
+        client = clientRegistry.getClient(user);
+        if (client == null) {
+            throw new TestHarnessInitializationException("Client '%s' has not been registered yet", user);
+        }
```
Comment (kjetilk) [inline src/main/java/org/solid/testharness/http/SolidClient.java:43]: I don't understand what role this plays with respect to the goal of the PR :-)

**Review by kjetilk — COMMENTED** [review]
OK, so often, I simply don't understand stuff, so rather than let it pass, I'll just say so :-) I think I got the gist of the PR except for one thing that I didn't understand, see inline comment.

File: src/main/java/org/solid/testharness/http/SolidClient.java (hunk 1 above)
Comment (edwardsph) [reply src/main/java/org/solid/testharness/http/SolidClient.java:43]: It is something that was sensible to do even if not part of the PR, however it was essential to the tests which were within the intent of the PR. It allows me to mock clients that are used in tests.

Comment (kjetilk) [reply src/main/java/org/solid/testharness/http/SolidClient.java:43]: OK, sounds good.

**Review by kjetilk — APPROVED** [review]
Also good!

**Pull request merged**